lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1626,10 +1627,29 @@ int swift::performFrontend(ArrayRef<const char *> Args,
   llvm::sys::fs::current_path(workingDirectory);
 
   // Parse arguments.
-  if (Invocation.parseArgs(Args, Instance->getDiags(), workingDirectory)) {
+  SmallVector<std::unique_ptr<llvm::MemoryBuffer>, 4> configurationFileBuffers;
+  if (Invocation.parseArgs(Args, Instance->getDiags(),
+                           &configurationFileBuffers, workingDirectory)) {
     return finishDiagProcessing(1);
   }
 
+  // Make an array of PrettyStackTrace objects to dump the configuration files
+  // we used to parse the arguments.
+  //
+  // This can't be a SmallVector or similar because PrettyStackTraces can't be
+  // moved (or copied)...and it can't be an array of non-optionals because
+  // PrettyStackTraces can't be default-constructed. So we end up with a
+  // dynamically-sized array of optional PrettyStackTraces, which get
+  // initialized by iterating over the buffers we collected above.
+  auto configurationFileStackTraces = llvm::make_unique<Optional<PrettyStackTraceFileContents>[]>(
+        configurationFileBuffers.size());
+  for_each(configurationFileBuffers.begin(), configurationFileBuffers.end(),
+           &configurationFileStackTraces[0],
+           [](const std::unique_ptr<llvm::MemoryBuffer> &buffer,
+              Optional<PrettyStackTraceFileContents> &trace) {
+    trace.emplace(*buffer);
+  });
+
```

Comment:
I would be tempted to enclose 1629-1651 in curly braces to show the reader that `configurationFileBuffers` and `configurationFileStackTraces` were not used again in this function.

Reply:
Ah, yeah, I'd love to…but they actually need to stay alive, since the PrettyStackTrace implementation is a linked list with stack-based lifetimes.

Reply:
Did I overlook a reference to it down below? Or is it non-obvious?? If the latter, you could save me some potential grief in the future with a comment to that effect in the code. Otherwise, I would come along, add the braces, and mess things up!

Reply:
It's RAII. I think we don't bother usually because "PrettyStackTrace" is already something weird, but in this case that type is buried in the unique_ptr, so I can make the comment more verbose.

Reply:
Yes. I have a habit of grepping for a local identifier and isolating it in a block if I don't see it used later. Since I tend to skim, it's easy for me to miss the fact than something is RAII and needs to stay alive, perhaps also because we don't always isolate locals in blocks in long routines when they could be.